scaleway/resource_lb_route.go

```diff
@@ -39,10 +39,28 @@ func resourceScalewayLbRoute() *schema.Resource {
 				ValidateFunc: validationUUIDorUUIDWithLocality(),
 				Description:  "The backend ID destination of redirection",
 			},
-			"match_sni": {
-				Type:        schema.TypeString,
+			"match": {
+				Type:        schema.TypeList,
 				Optional:    true,
-				Description: "The domain to match against",
+				Description: "The value to match a redirection",
+				MaxItems:    1,
+				MinItems:    1,
+				Elem: &schema.Resource{
+					Schema: map[string]*schema.Schema{
+						"sni": {
```

Comment:
It may be better to take this out of the resource as a list though and just have a single option. The resource `lb` looks pretty complex or not clear enough to use with many certificates, many applications